util/stream_request.py

```diff
@@ -523,20 +529,22 @@ def _find_stream_same_node(stream_key, streams, stream_dictionary):
         :return: StreamKey if found, otherwise None
         """
         log.debug('_find_stream_same_node(%r, %r, STREAM_DICTIONARY)', stream_key, streams)
-        method = stream_key.method
         subsite = stream_key.subsite
         node = stream_key.node
 
         for stream in streams:
-            sensors = stream_dictionary.get(stream, {}).get(method, {}).get(subsite, {}).get(node, [])
-            if sensors:
-                return StreamKey.from_dict({
-                    "subsite": subsite,
-                    "node": node,
-                    "sensor": sensors[0],
-                    "method": method,
-                    "stream": stream
-                })
+            # do not restrict the method - try all available
+            potential_methods = stream_dictionary.get(stream, {}).keys()
+            for method in potential_methods:
+                sensors = stream_dictionary.get(stream, {}).get(method, {}).get(subsite, {}).get(node, [])
```

Comment:
Same issue as previous comment. Simply returning the first method that matches will lead to unexpected results.